### PR TITLE
[Ascend] use aclnnUnique2 impl unique

### PR DESCRIPTION
**Pull request by hellozmz**
## Motivation and Context

[Ascend] use aclnnUnique2 impl unique

## Checklist
**Before PR**:

- [ ] I have read and followed the workflow indicated in the [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Review by zhangzefeng92 — APPROVED** [review]

File: impl/ascend/functions/unique.cpp [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+#include "../aclnn/adaptor.hpp"
+
+namespace impl {
+namespace ascend {
+
+diopiError_t diopiUnique(diopiContextHandle_t ctx, diopiTensorHandle_t* out, diopiConstTensorHandle_t input, const int64_t* dim, bool sorted, bool returnCounts,
+                         diopiTensorHandle_t indices, diopiTensorHandle_t* counts) {
+    // aclnnUnique2 only support dim == nullptr
+    ASCEND_CHECK_ABORT(dim == nullptr, "dim is not supported in aclnnUnique2");
```
Comment (jingguo-st) [inline impl/ascend/functions/unique.cpp:19]: 这种情况是不是对应传所有的dim进去就好了？

**Review by jingguo-st — APPROVED** [review]

File: impl/ascend/functions/unique.cpp [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
```
Comment (yangbofun) [inline impl/ascend/functions/unique.cpp:19]: 华为有aclnnUniqueDim来调用 带dim的unique.

Comment (hellozmz) [reply impl/ascend/functions/unique.cpp:19]: > 华为有aclnnUniqueDim来调用 带dim的unique. image

嗯嗯，尝试用aclnnUniqueDim去实现支持dim的场景，不过有些测试用例没有通过，看模型中暂时使用的是unique2，先实现了不使用dim的算子，并且用ASCEND_CHECK_ABORT检查了dim。

File: impl/ascend/functions/unique.cpp [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    if (dim) {
+        diopiRequireTensor(ctx, &outTmp, &inSize, nullptr, inputAt.dtype(), diopi_device);
+    } else {
+        diopiRequireTensor(ctx, &outTmp, &numelSize, nullptr, inputAt.dtype(), diopi_device);
+    }
+
+    // allocate temp inverse tensor
+    diopiTensorHandle_t inverseTmp = nullptr;
+    if (returnInverse || returnCounts) {
+        diopiRequireTensor(ctx, &inverseTmp, &inSize, nullptr, diopi_dtype_int64, diopi_device);
```
Comment (yangbofun) [inline impl/ascend/functions/unique.cpp:41]: 参考camb的requiresTensor写一些函数通用函数，这样就不用每次都从diopiSize_t来获取tensor

**Review by yewentao256 — APPROVED** [review]

**Review by yangbofun — APPROVED** [review]

**Pull request merged**